### DISCUSSION TITLE
Failed to bind socket: Address already in use

### DISCUSSION THREAD

**Discussion by cyberdynesoftware** [MusicPlayerDaemon/MPD#1724] [General]
I did struggle a bit to get mpd working, because mpd produces this error:

> Jan 14 15:46:07 butler mpd[2174]: server_socket: bind to ‘0.0.0.0:6600’ failed (continuing anyway, because binding to ‘[::]:6600’ succeeded): Failed to bind socket: Address already in use

Turns out that I had problems for other reasons: I think that the user in mpd.conf needs to match the user with which I start mpd with systemctl. Maybe this is obvious, but I stumbled on it.

Anyway, I got it working. Above error message persists in the log though. My understanding is this: mpd binds itself first to [::]:6600, which also binds to 0.0.0.0:6600 automatically. Afterwards, mpd tries to bind to 0.0.0.0:6600, which fails with above error. If this is true, the error message is useless and I would consider it a bug.

The problem with the message is that it got me hunting in the wrong direction when I struggled to get mpd working. In the end, it had nothing to do with port 6600.

**Comment by robindegen** [comment]
I agree that this is incredibly confusing and misleading. It took me hours to debug this issue while, in fact, there was no issue at all. This message should really be improved.

**Comment by jcorporation** [comment]
This was already discussed: https://github.com/MusicPlayerDaemon/MPD/pull/712

And it is mentioned in the documentation: https://mpd.readthedocs.io/en/stable/user.html#error-bind-to-0-0-0-0-6600-failed-continuing-anyway-because-binding-to-6600-succeeded

**Comment by herrrey** [comment]
Yeah, that error can definitely send you down the wrong path. I’ve run into something similar before — IPv6 binding to [::]:6600 can automatically cover 0.0.0.0:6600, so the second bind attempt throws that “address already in use” message even though things are technically fine. It’s confusing and doesn’t really help when you’re troubleshooting.

Glad you figured out the user mismatch in mpd.conf — that one can be easy to overlook.

Honestly, debugging stuff like this reminds me of long evenings tweaking configs while fueled by [good coffee](https://share.google/ZgaWKB6sLJUUfBz1l). I usually end up at a chill spot like AutoFuel Cafe to clear my head and rethink the problem — sometimes stepping away helps more than staring at logs for hours.

[a comment by an automated account is not shown]
